volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,690 @@
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="hivelist", plugin=hivelist.HiveList, version=(1, 0, 0)
+            ),
+        ]
+
+    def generate_timeline(
+        self,
+    ) -> Iterator[Tuple[str, timeliner.TimeLinerType, datetime.datetime]]:
+        for _, (
+            path,
+            _,
+            last_mod,
+            last_mod_2,
+            install_time,
+            compile_time,
+            _,
+            _,
+            _,
+            entry_type,
+        ) in self._generator():
+            if isinstance(last_mod, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} registry key modified", timeliner.TimeLinerType.MODIFIED, last_mod
+            if isinstance(last_mod_2, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} STANDARD_INFORMATION create time", timeliner.TimeLinerType.CREATED, last_mod_2
+            if isinstance(install_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} installed", timeliner.TimeLinerType.CREATED, install_time
+            if isinstance(compile_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} compiled (PE metadata)", timeliner.TimeLinerType.MODIFIED, compile_time
+
+    @classmethod
+    def get_amcache_hive(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel: interfaces.context.ModuleInterface,
+    ) -> Optional[registry.RegistryHive]:
+        """Retrieves the `Amcache.hve` registry hive from the kernel module, if it can be located."""
+        return next(
+            hivelist.HiveList.list_hives(
+                context=context,
+                base_config_path=interfaces.configuration.path_join(
+                    config_path, "hivelist"
+                ),
+                layer_name=kernel.layer_name,
+                symbol_table=kernel.symbol_table_name,
+                filter_string="amcache",
+            ),
+            None,
+        )
+
+    @classmethod
+    def parse_file_key(
+        cls, file_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[NullableString, AmcacheEntry]]:
+        """Parses File entries from the Windows 8 `Root\\File` key.
+
+        :param programs_key: The `Root\\File` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\Program` entries, and the second member is the `AmcacheEntry`.
+        """
+
+        def _handle_timestamp(value) -> NullableDatetime:
+            return (
+                conversion.wintime_to_datetime(value)
+                if isinstance(value, int)
+                else (
+                    renderers.NotAvailableValue()
```

Comment:
Ok, distinguishing between these two states does seems worthwhile, so I'm happy to be easier on using them/passing them around.

Reply:
I was trying to find a good way to differentiate on the things that are 'definitively missing' vs things that have an unexpected value type or could not be recovered due to `InvalidPagedAddressException`s and I'm still not 100% sure I'm happy with this since, as you said, it feels like it would be best to let the consumers of these things deal with the renderers.